chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -525,46 +526,95 @@ func (b *BitcoindNotifier) confDetailsManually(confRequest chainntnfs.ConfReques
 	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation,
 	chainntnfs.TxConfStatus, error) {
 
-	// Begin scanning blocks at every height to determine where the
-	// transaction was included in.
+	// batchRequest will store the scan requests at every height to determine
+	// where the transaction was included in.
+	batchRequest := make(
+		map[uint32]rpcclient.FutureGetBlockHashResult, currentHeight-heightHint,
+	)
+
 	for height := currentHeight; height >= heightHint && height > 0; height-- {
+		sendReq, err := b.chainConn.GetBlockHashAsync(int64(height))
+		if err != nil {
+			return nil, chainntnfs.TxNotFoundManually,
+				fmt.Errorf("unable to receive the block "+
+					"promise %v", err)
+		}
+
+		batchRequest[height] = sendReq
+	}
+
+	// Sending the bulk request using the updated batchClient.
+	b.chainConn.SendAsyncQueue()
+
+	blockHashAddresses := make([]*chainhash.Hash, 0, len(batchRequest))
+
+	for height, response := range batchRequest {
 		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+		// processing the next height response.
 		select {
 		case <-b.quit:
 			return nil, chainntnfs.TxNotFoundManually,
 				chainntnfs.ErrChainNotifierShuttingDown
 		default:
 		}
 
-		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		// Receive the next block hash from the async queue.
+		blockHash, err := response.Receive()
 		if err != nil {
 			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get hash from block "+
-					"with height %d", height)
+				fmt.Errorf("unable to retrieve hash for block "+
+					"with height %d: %v", height, err)
 		}
 
-		block, err := b.GetBlock(blockHash)
-		if err != nil {
+		blockHashAddresses = append(blockHashAddresses, blockHash)
```

Comment:
At this point the order of the block hashes will depend on the order of iteration of the `batchRequests` map (which is unordered so undefined). Again, testing is super important for code like this as it's easy to miss such bugs. (Same problem in `historicalSpendDetails`).

Reply:
Yes thats a great catch, Coming from C++ background I looked if golang supports something like ordered_map but there is only unordered_map so found few alternative ways like creating slice of keys and sort the slice then traverse over the map with the sorted keys thus maintaining the order.

Here since I already know the starting point of `height` which is `currentHeight` and it will go down till `heightHint` I will traverse accordingly to maintain the order.